packages/remix/src/utils/web-fetch.ts

```diff
@@ -0,0 +1,122 @@
+ */
+const getInternalSymbols = (
+  request: Record<string, unknown>,
+): {
+  bodyInternalsSymbol: string;
+  requestInternalsSymbol: string;
+} => {
+  const symbols = Object.getOwnPropertySymbols(request);
+  return {
+    bodyInternalsSymbol: symbols.find(symbol => symbol.toString().includes('Body internals')) as any,
+    requestInternalsSymbol: symbols.find(symbol => symbol.toString().includes('Request internals')) as any,
+  };
+};
+
+/**
+ * Vendored from:
+ * https://github.com/remix-run/web-std-io/blob/f715b354c8c5b8edc550c5442dec5712705e25e7/packages/fetch/src/utils/get-search.js#L5
+ */
+export const getSearch = (parsedURL: URL): string => {
+  if (parsedURL.search) {
+    return parsedURL.search;
+  }
+
+  const lastOffset = parsedURL.href.length - 1;
+  const hash = parsedURL.hash || (parsedURL.href[lastOffset] === '#' ? '#' : '');
+  return parsedURL.href[lastOffset - hash.length] === '?' ? '?' : '';
+};
+
+/**
+ * Convert a Request to Node.js http request options.
+ * The options object to be passed to http.request
+ * Vendored / modified from:
+ * https://github.com/remix-run/web-std-io/blob/f715b354c8c5b8edc550c5442dec5712705e25e7/packages/fetch/src/request.js#L259
+ */
+export const normalizeRemixRequest = (request: RemixRequest): Record<string, any> => {
+  const { requestInternalsSymbol, bodyInternalsSymbol } = getInternalSymbols(request);
+
+  if (!requestInternalsSymbol) {
+    throw new Error('Could not find request internals symbol');
+  }
+
+  const { parsedURL } = request[requestInternalsSymbol];
+  const headers = new Headers(request[requestInternalsSymbol].headers);
+
+  // Fetch step 1.3
+  if (!headers.has('Accept')) {
+    headers.set('Accept', '*/*');
+  }
+
+  // HTTP-network-or-cache fetch steps 2.4-2.7
+  let contentLengthValue = null;
+  if (request.body === null && /^(post|put)$/i.test(request.method)) {
+    contentLengthValue = '0';
+  }
+
+  if (request.body !== null) {
+    const totalBytes = request[bodyInternalsSymbol].size;
+    // Set Content-Length if totalBytes is a number (that is not NaN)
+    if (typeof totalBytes === 'number' && !Number.isNaN(totalBytes)) {
+      contentLengthValue = String(totalBytes);
+    }
+  }
+
+  if (contentLengthValue) {
+    headers.set('Content-Length', contentLengthValue);
+  }
+
+  // HTTP-network-or-cache fetch step 2.11
+  if (!headers.has('User-Agent')) {
+    headers.set('User-Agent', 'node-fetch');
+  }
+
+  // HTTP-network-or-cache fetch step 2.15
+  if (request.compress && !headers.has('Accept-Encoding')) {
+    headers.set('Accept-Encoding', 'gzip,deflate,br');
+  }
+
+  let { agent } = request;
+
+  if (typeof agent === 'function') {
+    agent = agent(parsedURL);
+  }
+
+  if (!headers.has('Connection') && !agent) {
+    headers.set('Connection', 'close');
+  }
+
+  // HTTP-network fetch step 4.2
+  // chunked encoding is handled by Node.js
+  const search = getSearch(parsedURL);
+
+  // Manually spread the URL object instead of spread syntax
```

Comment:
Is this your choice or just copied from their implementation? (Just curious why the spread operator is being avoided.)

Same question with the `@ts-ignore`s below.

Reply:
Neither of the above is a blocker (just my own curiosity), so I'm going to go ahead and merge this.

Reply:
Yes, they are copied from the [original implementation](https://github.com/remix-run/web-std-io/blob/f715b354c8c5b8edc550c5442dec5712705e25e7/packages/fetch/src/request.js#L311). Not sure what's the reason for spread operator there, but it seems they have kept [the approach of `node-fetch`](https://github.com/node-fetch/node-fetch/blob/55a4870ae5f805d8ff9a890ea2c652c9977e048e/src/request.js#L302).